src/request/body.rs

```diff
@@ -174,14 +174,13 @@ impl Payload for DataAndTrailers {
     }
 
     #[inline]
-    fn copy_to_vec(mut self) -> Vec<u8> {
-        let mut dest = Vec::with_capacity(self.len());
+    fn copy_to_bytes(self) -> Bytes {
+        let mut dest = BytesMut::with_capacity(self.len());
 
-        for frame in &mut self.frames {
+        for frame in &self.frames {
             dest.extend_from_slice(frame);
```

Comment:
`extend_from_slice` is used because each byte has to be copied into dest regardless of whether or not frame is owned. Working with a slice avoids an atomic operation for each frame in the payload.